recipe_recommender.mdx

```diff
@@ -0,0 +1,395 @@
+from suggestion import suggest_final_recipe
+
+PINECONE_API_KEY = os.environ.get('PINECONE_API_KEY')
+
+def get_recipes_from_ids(recipe_ids):
+    """Retrieve recipe contents for a given list of recipe IDs"""
+    results = []
+    filepaths = {DATA_DIR / f'{recipe_id}.txt' for recipe_id in recipe_ids}
+
+    for filepath in filepaths:
+        content = read_txt_file(filepath)
+        results.append(content)
+
+    return results
+
+def get_related_recipe_ids_for_text(text):
+    """Fetch IDs of recipes related to the given text using Pinecone"""
+    index = pinecone.Index(PINECONE_INDEX_NAME)
+    vector = get_vector_for_text(text)
+    query_results = index.query(
+        vector=vector,
+        top_k=2,
+        include_values=True
+    )
+    return {i['id'] for i in query_results['matches']}
+
+def main():
+    prompt = 'I like chicken.'
+    recipe_ids = get_related_recipe_ids_for_text(prompt)
+    recipes_list = get_recipes_from_ids(recipe_ids)
+    suggest_final_recipe(prompt, recipes_list)
+
+if __name__ == '__main__':
+    pinecone.init(api_key=PINECONE_API_KEY, environment='gcp-starter')
+    main()
+```
+</CodeGroup>
+
+After running this script you will be able to see the results in PromptLayer.
+
+![Recipe Request Log Screenshot](images/recipe-request-log.png)
+
+# Improving Results with Prompt Engineering
+
+Things are working, but let's tweak our prompt a bit to see if we can get some better results. Instead of `I like
+{food}`, try using the text `My favorite food is {food}`.
+
+This is a simple example of **prompt engineering**. The input we feed into OpenAI often needs to be tinkered with in
+order to get the best results. And once we find an input that works well, we don't want to lose it, even as we continue
+trying different prompts. Wouldn't it be nice if there was an easy way to keep track of all these prompts? Well lucky
+for us, there is.
+
+The Prompt Registry allows you to easily manage your prompt templates, which are customizable prompt strings with
+placeholders for variables. Specifically, a prompt template is your prompt string with variables indicated in curly
+brackets (`This is a prompt by {author_name}`).
+
+### Creating prompt templates
+
+Let's create a prompt template for our recipe app. In a new python file run the following script.
+
+```python
+import os
+
+import promptlayer
+
+openai = promptlayer.openai
+openai.api_key = os.getenv('OPENAI_API_KEY')
+promptlayer.api_key = os.environ.get('PROMPTLAYER_API_KEY')
+
+template = {
+    'input_variables': ['food', 'recipes_string'],
+    'template': 'I like {food}\n\n{recipes_string}',
+    'template_format': 'f-string',
+}
+
+promptlayer.prompts.publish("recipe_template", prompt_template=template)
+```
+
+After running it you will be able to see your prompt template in the Prompt Registry.
+
+![Recipe Prompt Registry Screenshot](images/recipe-prompt-registry.png)
+
+Here is an example of how you would create it in the UI.
+
+<video controls="controls">
```

Comment:
Make a new video with this example :)

Whole point of an example is being able to have more realistic screenshots and videos